src/pages/createCostModelWizard/table.tsx

```diff
@@ -57,38 +59,44 @@ const SourcesTable: React.SFC<InjectedTranslateProps> = ({ t }) => {
               </TextContent>
             </StackItem>
             <StackItem>
-              <Toolbar>
-                <ToolbarSection
-                  aria-label={t(
-                    'cost_models_wizard.source_table.filter_section_aria_label'
-                  )}
-                >
-                  <FilterToolbar isSingleOption />
-                  <ToolbarGroup style={{ marginLeft: 'auto' }}>
-                    <ToolbarItem>
-                      <Pagination
-                        isCompact
-                        itemCount={sources.length}
-                        perPage={perPage}
-                        page={page}
-                        onSetPage={(_evt, newPage) => {
-                          fetchSources(type, query, newPage, perPage);
-                        }}
-                        onPerPageSelect={(_evt, newPerPage) =>
-                          fetchSources(type, query, 1, newPerPage)
-                        }
-                      />
-                    </ToolbarItem>
-                  </ToolbarGroup>
-                </ToolbarSection>
-                <ToolbarSection
-                  aria-label={t(
-                    'cost_models_wizard.source_table.toolbar_results_section'
-                  )}
-                >
-                  <FilterResults />
-                </ToolbarSection>
-              </Toolbar>
+              <AssignSourcesToolbar
+                t={t}
+                filter={{
+                  onRemove: (category, chip) =>
+                    fetchSources(
+                      type,
+                      removeMultiValueQuery(query)(category, chip),
+                      1,
+                      perPage
+                    ),
+                  onClearAll: () => fetchSources(type, {}, 1, perPage),
+                  query,
+                }}
+                searchInputProps={{
+                  id: 'assign-source-search-input',
+                  value: filterName,
+                  onChange: onFilterChange,
+                  onSearch: _evt => {
+                    fetchSources(
+                      type,
+                      addMultiValueQuery(query)('name', filterName),
+                      1,
+                      perPage
+                    );
+                  },
+                }}
+                paginationProps={{
+                  isCompact: true,
+                  itemCount: sources.length,
+                  perPage,
+                  page,
+                  onSetPage: (_evt, newPage) => {
+                    fetchSources(type, query, newPage, perPage);
+                  },
+                  onPerPageSelect: (_evt, newPerPage) =>
+                    fetchSources(type, query, 1, newPerPage),
+                }}
```

Comment:
Just a suggestion... Instead of providing component props, you could provide the pagination component itself. This helped make my toolbar a little cleaner, didn't need to provide `t`, etc. Then, the table can interact with the pagination component's callbacks directly.